Sources/MainUnit.pas

```diff
@@ -203,6 +208,77 @@ procedure TMainForm.AboutActionExecute(Sender: TObject);
   AboutBox.ShowModal;
 end;
 
+procedure TMainForm.actLoadPGNFileExecute(Sender: TObject);
+var
+  SList : TStringList;
+  i, R, j, row : Integer;
+  s, s2, pgntags, white, black : AnsiString;
+  RegexObj: TRegExpr;
+begin
+  //load pgn file
+  RegexObj := TRegExpr.Create();
+  RegexObj.Expression := '((((\[.*?\])\s*)+)(1\..+?(0-1|1-0|1/2-1/2)))';
+  if dlgOpenPGN.Execute() then
+  begin
+    frmPGNDatabase.Show();
+    frmPGNDatabase.gridPGNDatabase.Clear();
+    frmPGNDatabase.gridPGNDatabase.RowCount := 1 + frmPGNDatabase.gridPGNDatabase.FixedRows;
+    s := '';
+    s2 := '';
+    R := 1;
+    row := frmPGNDatabase.gridPGNDatabase.FixedRows;
+    SList := TStringList.Create();
+    SList.LoadFromFile(dlgOpenPGN.FileName);
+    for i := 0 to SList.Count-1 do
+    begin
+      s2 := SList[i];
+      s += s2 + #13#10;
+       if (s2.Contains(' 1-0') or s2.Contains(' 1/2-1/2') or s2.Contains(' 0-1')) and RegexObj.Exec(s) then
+       begin
+         pgntags := RegexObj.Match[2];
+         j := pgntags.IndexOf('[White "');
+         white := '';
+         if (j > 0) then
+         begin
+           white := pgntags.Substring(j+8, pgntags.Substring(j+8).IndexOf('"]'));
+         end;
+         j := pgntags.IndexOf('[Black "');
+         black := '';
+         if (j > 0) then
+         begin
+           black := pgntags.Substring(j+8, pgntags.Substring(j+8).IndexOf('"]'));
+         end;
+
+         frmPGNDatabase.gridPGNDatabase.RowCount := frmPGNDatabase.gridPGNDatabase.RowCount + 1;
+
+         frmPGNDatabase.gridPGNDatabase.Cells[0, row] := RegexObj.Match[2];
+         frmPGNDatabase.gridPGNDatabase.Cells[1, row] := white;
+         frmPGNDatabase.gridPGNDatabase.Cells[2, row] := black;
+         frmPGNDatabase.gridPGNDatabase.Cells[3, row] := RegexObj.Match[5];
+         Inc(row);
+         s := '';
+         R := R + 1;
+       end
+    end;
+    if (R-1 = 1) then
+    begin
+       frmPGNDatabase.Caption := FormatFloat('#,',R-1) + ' PGN Game';
+    end else
+    begin
+       frmPGNDatabase.Caption := FormatFloat('#,',R-1) + ' PGN Games';
+    end;
+
+    try
+       CreateNewGame();
```

Comment:
AFAIR `CreateNewGame` will just create the new game with the current new game dialog settings. So, if you are not in the analysis mode now, then the following will happen:

* the current game will be restarted (not in analysis mode)
* in the game mode, all the attempts to replace the current notation will be disallowed

So, it's necessary to disallow using this in the analysis mode.